documentation/design-docs/dotnet-tools.md

```diff
@@ -398,6 +399,49 @@ CONVERT
       Writing:       ./trace.speedscope.json
       Conversion complete
 
+REPORT
+
+    dotnet-trace report [-h|--help]
+                        [-t|--type]
+                        [-p|--parameters]
+                        [-o|--output <output_file_path>]
+                        --header
+                        <trace_file>
+
+    Generate report into stdout from a previously generated trace_file.
+
+    -h, --help
+        Show command line help
+    
+    -t, --type
+        Specifies the type of report to create. The first type of report that will be available is 'topN'
+
+    -p, --parameters
+        Takes a list of arguments for each specific report type, see below for details.
+        type            parameters  explanation
+        'topN'          -n          the number of top functions to gather on the stack
+                        -i          find the top n methods inclusively on the stack. Without this tag, will find the top N methods exclusively on the stack
+        'countEvent'    
+
+    -o, --output
+        The path where the converted file is written. If unspecified, the file is written in the current directory
+        using the same base filename as the input file and the extension appropriate for the new format.
+
+    --header
+        Creates a header above the usual output containing general information about what was collected
```

Comment:
Sounds great, the reasoning behind the header was to pattern this tool after perf script.

Reply:
Ah, I thought the header flag was implying it controlled whether the header for the report, i.e., the field names above the columns, was printed. I think the header flag in `perf` adds a header that includes a bunch of metadata about the trace in addition to the report requested. I think we should include this option. I imagine it would render something like this:

```
dotnet trace report topn -n 5 --inclusive --header

Trace Duration: <timestamp>
Total Event Count: <>
Total Stack Samples: <>
Clock Speed: <>
OS: <>
Arch: <>

<rest of report>
```

Let me see if I can find some prior art on this.

Reply:
I agree with Brian. If we are going to have the header, we should have the flag to enable it. This way the default is easily consumable in a shell pipe without a "skip until <identifier>" in the script.